### PR TITLE
Fix RealtimePresence.get

### PR DESCRIPTION
**Pull request by ricardopereira**
#271

### CODE REVIEW

**Comment by ricardopereira** [comment]
```
[18:40:48]: ▸ Failing tests:
[18:40:48]: ▸ -[ARTRealtimePresenceTest testEnterAndGet]
[18:40:48]: ▸ -[ARTRealtimePresenceTest testEnterClient]
[18:40:48]: ▸ -[ARTRealtimePresenceTest testEnterClient]
[18:40:48]: ▸ -[ARTRealtimePresenceTest testLeaveAndGet]
[18:40:48]: ▸ -[ARTRealtimePresenceTest testPresenceNoSideEffects]
[18:40:48]: ▸ -[ARTRealtimePresenceTest testPresenceWithData]
[18:40:48]: ▸ -[ARTRealtimePresenceTest testPresenceWithData]
```

One of the problems is related with the order of the members in the `PresenceMap`. Sometimes the first connected client is in second place on the map. I looked at the specs and there's nothing specifying how it should be ordered`. Should it be ordered by `clientId` or respect the order they were added?

**Comment by ricardopereira** [comment]
@mattheworiordan @tcard ^

**Comment by tcard** [comment]
@ricardopereira I'd say we should leave it unspecified. I don't see how the order matters.

**Comment by ricardopereira** [comment]
@tcard Looks good?

File: Source/ARTChannel.h [hunk 1]
```diff
@@ -33,8 +33,8 @@ ART_ASSUME_NONNULL_BEGIN
 - (void)publish:(__GENERIC(NSArray, ARTMessage *) *)messages;
 - (void)publish:(__GENERIC(NSArray, ARTMessage *) *)messages callback:(art_nullable void (^)(ARTErrorInfo *__art_nullable error))callback;
 
-- (void)history:(void(^)(__GENERIC(ARTPaginatedResult, ARTMessage *) *__art_nullable result, NSError *__art_nullable error))callback;
-- (BOOL)history:(art_nullable ARTDataQuery *)query callback:(void(^)(__GENERIC(ARTPaginatedResult, ARTMessage *) *__art_nullable result, NSError *__art_nullable error))callback error:(NSError *__art_nullable *__art_nullable)errorPtr;
+- (void)history:(void(^)(__GENERIC(ARTPaginatedResult, ARTMessage *) *__art_nullable result, ARTErrorInfo *__art_nullable error))callback;
```
Comment (tcard) [inline Source/ARTChannel.h]: This change needs to be done in https://github.com/ably/docs as well.

Comment (ricardopereira) [inline Source/ARTChannel.h]: You mean the code examples?

File: Source/ARTRealtimePresence.m [hunk 2]
```diff
@@ -15,26 +15,65 @@
...
-    [[self channel] throwOnDisconnectedOrFailed];
-    [super get:query callback:callback];
+- (void)get:(void (^)(NSArray<ARTPresenceMessage *> *, ARTErrorInfo *))callback {
+    [self get:[[ARTRealtimePresenceQuery alloc] init] callback:callback];
+}
+
+- (void)get:(ARTRealtimePresenceQuery *)query callback:(void (^)(NSArray<ARTPresenceMessage *> *, ARTErrorInfo *))callback {
+    [self.channel throwOnDisconnectedOrFailed];
+    [self.channel attach:^(ARTErrorInfo *error) {
+        if (query.waitForSync) {
```
Comment (tcard) [inline Source/ARTRealtimePresence.m]: What about the rest of the members in ARTRealtimePresenceQuery?

By the way, according to [the spec](http://docs.ably.io/client-lib-development-guide/features/#idl), the query parameters for this method are `waitForSync`, `clientId` and `connectionId`. But our ARTRealtimePresenceQuery inherits from REST's ARTPresenceQuery, which has other parameters.

File: Source/ARTChannel.h (hunk 1 above)
Comment (tcard) [inline Source/ARTChannel.h]: Yes.

File: Tests/ARTRealtimePresenceTest.m [hunk 3]
```diff
@@ -486,18 +486,13 @@ -(void) testEnterAndGet {
...
-                    XCTAssertEqual(2, messages.count);
-                    ARTPresenceMessage *m0 = messages[0];
-                    ARTPresenceMessage *m1 = messages[1];
-                    XCTAssertEqual(m0.action, ARTPresencePresent);
-                    XCTAssertEqual(m1.action, ARTPresencePresent);
-                    //TODO work out why the order seems to be random
-                    //XCTAssertEqualObjects(enter2, [m0 data]);
-                    //XCTAssertEqualObjects(enter, [m1 data]);
+                    XCTAssertEqual(2, members.count);
+                    XCTAssertEqual(members[0].action, ARTPresencePresent);
```
Comment (tcard) [inline Tests/ARTRealtimePresenceTest.m]: Didn't you say here https://github.com/ably/ably-ios/pull/299#issuecomment-197005427 that the order isn't deterministic?

File: Source/ARTRealtimePresence.m (hunk 2 above)
Comment (ricardopereira) [inline Source/ARTRealtimePresence.m]: Oh, I see. So [ARTRestPresence.get](https://github.com/ably/ably-ios/blob/957b709f8e7cbb9ead8bfdd7d028ffd9e6826854/Source/ARTRestPresence.m#L31) is also missing those members, right? Maybe I will do it in another PR, WDYT?

Comment (tcard) [inline Source/ARTRealtimePresence.m]: Yeah, fine by me.

File: Tests/ARTRealtimePresenceTest.m (hunk 3 above)
Comment (ricardopereira) [inline Tests/ARTRealtimePresenceTest.m]: Yes. I forgot to remove this. :+1:

File: Source/ARTChannel.h (hunk 1 above)
Comment (ricardopereira) [inline Source/ARTChannel.h]: Issue #304

File: Source/ARTRealtimePresence.m (hunk 2 above)
Comment (ricardopereira) [inline Source/ARTRealtimePresence.m]: Issue #306

File: Tests/ARTRealtimePresenceTest.m (hunk 3 above)
Comment (ricardopereira) [inline Tests/ARTRealtimePresenceTest.m]: Done 4e80dc2.

**Pull request merged**